File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    sum(range(1, number + 1)) ** 2


def sum_of_squares(number):
    sum(map(lambda n: n ** 2, range(1, number + 1)))


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: c/difference-of-squares/src/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number) {
  unsigned int i, sum = 0;
  for(i = 1;i <= number;++i)
    sum += pow(i, 2);
  return sum;
}

unsigned int square_of_sum(unsigned int number) {
  unsigned int i, sum = 0;
  for(i = 1;i <= number;++i)
    sum += i;
  return pow(sum, 2);
}

unsigned int difference_of_squares(unsigned int number) {
  return square_of_sum(number) - sum_of_squares(number);
}
